app/src/main/java/de/danoeh/antennapod/preferences/PreferenceUpgrader.java

```diff
@@ -109,5 +109,12 @@ private static void upgrade(int oldVersion, Context context) {
         if (oldVersion < 2050000) {
             prefs.edit().putBoolean(UserPreferences.PREF_PAUSE_PLAYBACK_FOR_FOCUS_LOSS, true).apply();
         }
+        if (oldVersion < 2070000) {
+            // Migrate drawer feed counter setting to reflect removal of
+            // "unplayed and in inbox" (0), by changing it to "unplayed" (2)
+            if (UserPreferences.getFeedCounterSetting().id == 0) {
+                UserPreferences.setFeedCounterSetting("2");
```

Comment:
I would do the migration manually here, e.g., writing to the SharedPreferences like above. That way, we don't need to keep setters in the "main" code for things that are only ever used when upgrading. When accessing the SharedPreferences directly, you can also remove the enum because you can check the value here without having to "parse" it.

Reply:
OK, please see my latest changes and let me know if the way I did it was OK. I couldn't reference the preference key name using the  from UserPreferences because `PREF_DRAWER_FEED_COUNTER` is a private value.

Reply:
How about making the preference key public? I think most of the keys are public anyways.

Reply:
Sure, I will do that tonight and update this comment thread when done.

Reply:
Sorry, I got sidetracked the other day and wasn't able to do it until just now.

Reply:
Does this look OK to merge now?